src/callback_tracking.jl

```diff
@@ -205,6 +205,7 @@ function _setup_reverse_callbacks(cb::Union{ContinuousCallback, DiscreteCallback
     # if save_positions = [1,0] the gradient contribution is added before, and in principle we would need to correct the adjoint state again. Thefore,
 
     cb.save_positions == [1, 0] && error("save_positions=[1,0] is currently not supported.")
+    @assert sensealg.autojacvec isa Union{ReverseDiffVJP, EnzymeVJP}
```

Comment:
Instead of an assertion, it would be good to throw an informative error that only those two are allowed.